### PR TITLE
Feature/sticky-menu

### PR DESCRIPTION
**Pull request by MitskevichVS**
### Features ###
1. Added a safe interval for the header menu.
2. Fixed elastic scroll in safari.
### Screenshot ###
Nothing new to show

### CODE REVIEW

File: dist/js/header/headerController.js [hunk 1]
```diff
@@ -1,24 +1,100 @@
...
+      if (!this.scrollEventExecutionFlag) {
+        requestAnimationFrame(this.scrollEvent.bind(this));
+        this.scrollEventExecutionFlag = true;
+      }
+    });
+  }
+
+  scrollEvent() {
+    const scrollTopSafeSpace = 50; // 50px scroll top to show sticky menu
+    const docElement = document.documentElement;
```
Comment (danymill) [inline dist/js/header/headerController.js]: docElement should be initialized once on instance init

File: dist/js/header/headerController.js [hunk 2]
```diff
@@ -1,24 +1,100 @@
...
+    }
+
+    // Checking conditions for dynamic menu display
+    if (deltaOfPreviousAndCurrentCoords >= 0 && this.menuShowFlag) { // scroll top after menu was shown
+      this.headerMenuShouldShow();
+    } else if (this.previousYCoordinate <= scrollTopValueWithSafeInterval && pageYOffset >= this.headerMenuContainerSize) { // scroll bottom or check scroll more than default menu location
+      this.headerMenuShouldHide();
+    } else { // other
+      this.headerMenuShouldShow();
     }
```
Comment (danymill) [inline dist/js/header/headerController.js]: Looks like first `if` or last `else` is redundant

File: dist/js/header/headerController.js [hunk 3]
```diff
@@ -1,24 +1,100 @@
...
+
+  getHeaderElementSize() {
+    const elementHeight = parseInt(document.defaultView.getComputedStyle(this.headerMenuContainer, '').getPropertyValue('height'), 10);
+    const elementMargin = parseInt(document.defaultView.getComputedStyle(this.headerMenuContainer, '').getPropertyValue('top'), 10);
+
+    this.headerMenuContainerSize = elementHeight + elementMargin;
+  }
+
+  headerMenuShouldStick() {
+    if (this.headerMenuContainer.classList.contains('_stick')) {
```
Comment (danymill) [inline dist/js/header/headerController.js]: Working with classList can be optimized, please have a look at https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

**Review by danymill — CHANGES_REQUESTED** [review]

File: dist/js/header/headerController.js (hunk 1 above)
Comment (MitskevichVS) [reply dist/js/header/headerController.js]: fixed

File: dist/js/header/headerController.js (hunk 3 above)
Comment (MitskevichVS) [reply dist/js/header/headerController.js]: fixed

File: dist/js/header/headerController.js (hunk 2 above)
Comment (MitskevichVS) [reply dist/js/header/headerController.js]: fixed

**Review by danymill — APPROVED** [review]

**Pull request merged**